DESIGN.md

```diff
@@ -0,0 +1,60 @@
+# Bulk Data Extract (BDE) Processor design
+
+## User requirements
+
+-  The _user interface_ is a read-only PostgreSQL database in AWS RDS, with
+   Azure authentication
+-  _Accessible_ to all
+   [current BDE users](https://github.com/linz/ds-infra-team/wiki/BDE-users)
+-  Process BDE files
+   _[every day](https://toitutewhenua.atlassian.net/wiki/spaces/GEOD/pages/86418416/BDE+Processor+Product)
+   except Sunday_
+-  Optional _post-processing_ pulled in from separate internal repository
+-  All changes for a single day are imported within a _single transaction_
+-  _Immediate notification_ of completed import (like the legacy BDE Processor
+   SNS topic)
+
+## Technical requirements
+
+-  Thorough testing of every part of the process before merging any pull
+   requests
+-  Separate, managed accounts for infrastructure testing and production
+-  Trivial deployment
+
+## Open questions
+
+-  Can we give read access to the BDE data to the entire organisation? The
+   public?
+
+## Non-requirements
+
+-  Don't keep BDE files around. They should ideally be kept only in memory or at
+   most in temporary storage which is cleaned up after each run. If we
+   absolutely need to keep them around that should be done as a separate system.
+   If it's convenient, the BDE Processor could read from the data in that system
+   rather than doing its own data copying.
+-  No table versioning. It creates a _lot_ of extra complexity, and we don't
+   have any PostgreSQL experts at hand to maintain the extensions.
+
+## Wanted features
+
+-  Pull rather than push data from Landonline. This way we can keep the entire
+   system together, rather than having a costly and awkward split between
+   on-premise and cloud architecture.
+-  Continuous deployment. Any time a change is merged to the main branch some
+   smoke tests are performed and the result is deployed to production.
+-  React to change rather than run on a timer, where possible. Ideally the BDE
```

Comment:
This is how it works isnt it? As soon as the METADATA file (the last file) is written it kicks of the BDE processor. The reason it runs at approximately the same time every day, is because Landonline BDE unload is a scheduled job (although they do occasionally unload at other times).

To get around that Landonline scheduled job issue, then we would need something other than BDE files, like event streaming or direct DB access etc.